software/smartContract/contracts/OrderedIntervalList.sol

```diff
@@ -0,0 +1,91 @@
+pragma solidity ^0.4.24;
+
+/**
+    @title OrderedIntervalList
+    @dev List of ordered by intervals with non intersections checks. 
+ */
+library OrderedIntervalList {
+    struct Interval {
+        uint64 begin; // inclusive
+        uint64 end; // exclusive
+    }
+    struct Data {
+        Interval[] intervals;
+    }
+
+    function getSize(Data storage self) internal view returns(uint256 size) {
+        return self.intervals.length;
+    }
+
+    /**
+    @notice Get interval by list index
+    @param _index interval index in list
+    @return interval tuple
+     */
+    function get(Data storage self, uint _index) internal view returns(uint64 begin, uint64 end) {
+        require(self.intervals.length > 0 && _index < self.intervals.length, "check list is non empty and index have proper bounds");
+        Interval memory interval = self.intervals[_index];        
```

Comment:
s/memory/storage/ ?

Reply:
Maybe better to return whole struct, compile will split it automagically.